File: poeta.rb
#!/usr/bin/env ruby -wKU
# poeta.rb
#   juanfc 2010-04-14
# 

kCuantos= 5e3
kLength= 8192
$ctr= 0
now= Time.now

$words=<<END.downcase.split
Con diez cañones por banda viento en popa a toda vela
no corta el mar si no vuela un velero bergantín bajel
pirata llamado por su bravura el temido en todo el mar
conocido del uno al otro confín
END

def poema(length)
  r=""
  prev=""
  curr=""
  while r.length < length do
    curr=$words[rand($words.length)] until curr != prev
    r+= (prev=curr)+ " "
  end  
  $ctr+=1
  return r
end

while $ctr < kCuantos do
  poema(kLength)
end


puts(kCuantos.to_s + " poemas de "+ kLength.to_s+ " caracteres.");
puts((now=Time.now - now).to_s + " segundos.");
puts((kCuantos/now).to_s+ " poemas / segundo.");

File: poeta.js
#!/usr/bin/env node
//20100412 jorge@jorgechamorro.com

var kCuantos= 5e3;
var kLength= 16*1024;
var ctr= 0;
var now= +new Date();
var puts= print;

function rnd (n) {
  return (n* Math.random()) >>> 0;
}

var words= "Con diez cañones por banda viento en popa a toda vela no corta el mar si no vuela un velero bergantín bajel pirata llamado por su bravura el temido en todo el mar conocido del uno al otro confín".toLowerCase().split(" ");

function poeta (length, poema, curr, prev) {
  poema= "";
  while (poema.length < length) {
    do {
      curr= words[rnd(words.length)];
    } while (curr === prev);
    poema+= (prev= curr)+ " ";
  }
  ctr++;
  return poema;
}

var poemas= [];
while (ctr < kCuantos) {
  //poemas.push(poeta(kLength));
  poeta(kLength);
}

Number.prototype.to_s= function (p) {
  return (this / 1e3).toFixed(p);
};

puts(kCuantos+ " poemas de "+ kLength+ " caracteres.");
puts((now= (new Date()- now).to_s(1))+ " segundos.");
puts((kCuantos/ now).toFixed(1)+ " poemas / segundo.");
